ui/appui-react/src/appui-react/utils/UIError.ts

```diff
@@ -0,0 +1,15 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import type { GetMetaDataFunction } from "@itwin/core-bentley";
+import { BentleyError, BentleyStatus } from "@itwin/core-bentley";
+
+/** iTwin.js UI UiError class is a subclass of BentleyError, extended with a category. Errors are logged.
+ * @public
+ */
+export class UiError extends BentleyError {
```

Comment:
How is the `category` used? Is the expectation for `category` to be used while logging the errors (i.e. logger checks if an error is `instanceof UiError`)? Having multiple classes isn't a good approach if that's the case. Maybe we should consider either keeping a single class OR the need for category/UiError altogether?

Reply:
Right. My initial thinking was to eliminate the `UiError` altogether, but thought "it had to be used for something." However, I've found no evidence of loggers or other error handlers checking if the error is `UiError` or if the error has a category property. Certainly not used in core, and I found no usage in our most popular apps. 

I can ask around to see if anyone remembers the intent, but that is often a fruitless task.

Reply:
Remembered history is that it was just used in appui for all error types, but once logging categories were introduced, the `UiError.category` was essentially abandoned. I'll look to remove it altogether if possible.

Reply:
If we want to be safe, we cannot remove `UiError`. It's public and consumers could possible be using the category property even though I've found no evidence so far. I feel uncomfortable removing it from appui.

What do you think about deprecating it in `core/appui-abstract` (recommending `BentleyError` as a replacement), continuing to use it in appui until core@6 comes out, then remove it completely?

Reply:
Yes, I think that makes more sense, instead of re-introducing it once again.

Reply:
Reverted. I believe we can move to `BentleyError` at the next major for `AppUi`, however, since we wouldn't be deprecating any public API first.